src/connectivity/mod.rs

```diff
@@ -366,24 +379,42 @@ pub fn digraph_adjacency_matrix(
 ///     for all edges.
 /// :param float default_weight: If ``weight_fn`` is not used this can be
 ///     optionally used to specify a default weight to use for all edges.
+/// :param float null_value: An optional float that will treated as a null
+///     value. This is the default value in the output matrix and it is used
+///     to indicate the absence of an edge between 2 nodes. By default this is
+///     ``0.0``.
 ///
 /// :return: The adjacency matrix for the input dag as a numpy array
 /// :rtype: numpy.ndarray
-#[pyfunction(default_weight = "1.0")]
-#[pyo3(text_signature = "(graph, /, weight_fn=None, default_weight=1.0)")]
+#[pyfunction(default_weight = "1.0", null_value = "0.0")]
+#[pyo3(
+    text_signature = "(graph, /, weight_fn=None, default_weight=1.0, null_value=0.0)"
+)]
 pub fn graph_adjacency_matrix(
     py: Python,
     graph: &graph::PyGraph,
     weight_fn: Option<PyObject>,
     default_weight: f64,
+    null_value: f64,
 ) -> PyResult<PyObject> {
     let n = graph.node_count();
-    let mut matrix = Array2::<f64>::zeros((n, n));
+    let mut matrix = if null_value != 0.0 {
+        Array2::<f64>::from_elem((n, n), null_value)
+    } else {
+        Array2::<f64>::zeros((n, n))
+    };
```

Comment:
Update initialization here and in `distance_matrix.rs` as well?

Reply:
Oh yeah good catch, I only updated it the one place you had the suggestion before. Updated the others here: https://github.com/Qiskit/retworkx/pull/410/commits/d58c5c3e27b8241bc21da72634c25a69aff96de2